das/restful_client.go

```diff
@@ -40,6 +40,10 @@ func NewRestfulDasClientFromURL(url string) (*RestfulDasClient, error) {
 }
 
 func (c *RestfulDasClient) GetByHash(ctx context.Context, hash []byte) ([]byte, error) {
+	if len(hash) < 32 {
```

Comment:
Should we require `len(hash) == 32`?  It seems like a hash longer than 32 bytes probably reflects an error somewhere.

Reply:
There was a bug around extra hash length caused by base64 padding that I fixed now so I can make this an equality check.